apps/guide/content/docs/legacy/interactions/modals.mdx

```diff
@@ -35,112 +32,330 @@ client.on(Events.InteractionCreate, async (interaction) => {
 
-		// Add components to modal
-
-		// Create the text input components
-		const favoriteColorInput = new TextInputBuilder()
-			.setCustomId('favoriteColorInput')
-			// The label is the prompt the user sees for this input
-			.setLabel("What's your favorite color?")
-			// Short means only a single line of text
-			.setStyle(TextInputStyle.Short);
-
+		// [!code ++:6]
 		const hobbiesInput = new TextInputBuilder()
 			.setCustomId('hobbiesInput')
+			// Short means a single line of text.
+			.setStyle(TextInputStyle.Short)
+			// Placeholder text displayed inside the text input box
+			.setPlaceholder('card games, films, books, etc.');
+
+		// [!code focus:10]
+		const hobbiesLabel = new LabelBuilder()
+			// The label is a large header that identifies the interactive component for the user.
 			.setLabel("What's some of your favorite hobbies?")
-			// Paragraph means multiple lines of text.
-			.setStyle(TextInputStyle.Paragraph);
+			// The description is optional small text beneath the label that provides additional information about the interactive component.
+			.setDescription('Activities you like to participate in')
+			// [!code ++:2]
+			// Set text input as the component of the label
+			.setTextInputComponent(hobbiesInput);
+
+		// Add the label to the modal
+		modal.addLabelComponents(hobbiesLabel);
+	}
+});
+```
+
+#### Input styles
+
+Discord offers two different input styles:
+
+- `Short`, a single-line text entry
+- `Paragraph`, a multi-line text entry
+
+#### Input properties
+
+A text input field can be customized in a number of ways to apply validation or set default values via the following `TextInputBuilder` methods:
+
+```js
+const input = new TextInputBuilder()
+	// Set the component id (this is not the custom id)
+	.setId(0)
+	// Set the maximum number of characters allowed
+	.setMaxLength(1_000)
+	// Set the minimum number of characters required for submission
+	.setMinLength(10)
+	// Set a default value to pre-fill the text input
+	.setValue('Default')
+	// Require a value in this text input field (defaults to true)
+	.setRequired(true);
+```
 
-		// An action row only holds one text input,
-		// so you need one action row per text input.
-		const firstActionRow = new ActionRowBuilder().addComponents(favoriteColorInput);
-		const secondActionRow = new ActionRowBuilder().addComponents(hobbiesInput);
+<Callout>
+	The `id` field is used to differentiate components within interactions (which text input, selection, etc.). In
+	contrast, the `customId` covered earlier identifies the interaction (which modal, command, etc.).
+</Callout>
+
+### Select menu
+
+Select menus allow you to limit user input to a preselected list of values. Discord also offers selections for Discord-specific concepts like users, roles, and channels.
+Since they behave very similar as in messages, please refer to the [corresponding guide page](../interactive-components/select-menus) for more information on configuring select menus.
```

Comment:
```suggestion
Since they behave very similarly to how they do in messages, please refer to the [corresponding guide page](../interactive-components/select-menus) for more information on configuring select menus.
```